### ISSUE TITLE
Built-in iconv not detected in openSUSE Tumbleweed

### ISSUE THREAD

**Issue by arjendekorte** [Cisco-Talos/clamav#224]
The following is logged when attempting to build clamav-0.104.0-rc with openSUSE Tumbleweed:

```[    6s] -- Performing Test Iconv_IS_BUILT_IN  
[    6s] -- Performing Test Iconv_IS_BUILT_IN - Failed  
[    6s] CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):  
[    6s]   Could NOT find Iconv (missing: Iconv_LIBRARY)  
[    6s] Call Stack (most recent call first):  
[    6s] /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)  
[    6s]   cmake/FindIconv.cmake:129 (find_package_handle_standard_args)  
[    6s]   CMakeLists.txt:332 (find_package)
```  

Apparently this test is not able to determine if iconv is actually built-in. Adding '-D Iconv_IS_BUILT_IN=ON' to the cmake command line, works around this.

**Comment by arjendekorte** [comment]
`#nm -gD /usr/lib64/libc.so.6|grep iconv`

0000000000027ec0 T iconv
00000000000280b0 T iconv_close
0000000000027e10 T iconv_open

**Comment by micahsnyder** [comment]
Thanks for submitting the bug report. We'll find a proper fix for it before the stable release.  Once we have a PR to fix it, I'd love your help verifying that it works for you, especially since the failures I saw a moment ago on Leap were a little different.

**Comment by arjendekorte** [comment]
I suspect this might have something to do with the GCC version used. Tumbleweed uses GCC-11, while Leap 15.2 and 15.3 are still on GCC-7.

**Comment by arjendekorte** [comment]
cmake/FindIconv.cmake: fix return type in test code #233

**Comment by micahsnyder** [comment]
Merged (fixed) in main and cherry-picked the fix rel/0.104
